proto/osmosis/superfluid/query.proto

```diff
@@ -74,3 +122,82 @@ message ConnectedIntermediaryAccountRequest { uint64 lock_id = 1; }
 message ConnectedIntermediaryAccountResponse {
   SuperfluidIntermediaryAccountInfo account = 1;
 }
+
+message SuperfluidDelegationAmountRequest {
+  string delegator_address = 1;
+  string validator_address = 2;
+  string denom = 3;
+}
+
+message SuperfluidDelegationAmountResponse {
+  repeated cosmos.base.v1beta1.Coin amount = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+}
+
+message SuperfluidDelegationsByDelegatorRequest {
+  string delegator_address = 1;
+}
+
+message SuperfluidDelegationsByDelegatorResponse {
+  repeated SuperfluidDelegationRecord superfluid_delegation_records = 1
+      [ (gogoproto.nullable) = false ];
+  repeated cosmos.base.v1beta1.Coin total_delegated_coins = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+}
+
+message SuperfluidDelegationsByValidatorDenomRequest {
+  string validator_address = 1;
+  string denom = 2;
+}
+
+message SuperfluidDelegationsByValidatorDenomResponse {
+  repeated SuperfluidDelegationRecord superfluid_delegation_records = 1
+      [ (gogoproto.nullable) = false ];
+  repeated cosmos.base.v1beta1.Coin total_delegated_coins = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+}
```

Comment:
I suspect this query may cause problems if we don't add in pagination params.

Why do we need to get all delegations for a validator? (Does any staking dashboard meaningfully use this today? Mintscan's isn't great, and thats already indexed in a second table which can be done otherwise)

Reply:
Can we move the total to a second query / just use the other query